go/vt/discovery/healthcheck.go

```diff
@@ -531,6 +540,7 @@ func (hc *HealthCheckImpl) checkHealthCheckTimeout() {
 			hcc.mu.Unlock()
 			continue
 		}
+		hcc.streamCancelFunc()
```

Comment:
nits:

- Can you please add a newline before this line?
- Can you please add a comment, something like this: "Timeout detected. Cancel the current streaming RPC and let checkConn() restart it."